app/src/main/java/com/arctouch/codechallenge/preference/MainPreference.kt

```diff
@@ -0,0 +1,24 @@
+package com.arctouch.codechallenge.preference
+
+import android.content.Context
+import android.preference.PreferenceManager
+
+object MainPreference {
```

Comment:
Could you explain the motivation behind using shared prefs to pass the search query?

Reply:
To be honest again, there is no good reason for this, it was late at night, I had to deliver the challenge and work the other day, I had some trouble passing the String to the SearchViewModel, and only then to the SearchDataSourceFactory, so the fastest way was this. But I am fully aware that it is not the best choice, both from a performance point of view, and the modularity and maintainability of the code. On a normal day, I would not do that.